modules/face/src/facemarkLBF.cpp

```diff
@@ -535,13 +535,14 @@ Rect FacemarkLBFImpl::getBBox(Mat &img, const Mat_<double> shape) {
 void FacemarkLBFImpl::prepareTrainingData(Mat img, std::vector<Point2f> facePoints,
     std::vector<Mat> & cropped, std::vector<Mat> & shapes, std::vector<BBox> &boxes)
 {
+    Mat shape;
+    Mat _shape = Mat(facePoints).reshape(1);
+    Rect box = getBBox(img, _shape);
+
     if(img.channels()>1){
         cvtColor(img,img,COLOR_BGR2GRAY);
```

Comment:
Perhaps, it makes sense to convert color of cropped image only. Update the code below:

```
Mat crop = img(Rect((int)min_x, (int)min_y, (int)w, (int)h)).clone();
```

(BTW, avoid using of in-place src/dst parameters - they cause performance regressions)

Reply:
Oh, I completely agree... as you can see I just moved some existing code around (i.e. tried to change as little as possible).

Reply:
I believe it is OK to make API more friendly for Python or other bindings.